lib/compress/index.js

```diff
@@ -2436,7 +2432,7 @@ def_optimize(AST_Call, function(self, compressor) {
     }
 
     var ev = self.evaluate(compressor);
-    if (ev !== self) {
+    if (ev !== self && ev !== nullish) {
```

Comment:
We can remove these checks, by making `evaluate` return `this` when it sees `nullish`.

https://github.com/terser/terser/pull/1062/files#diff-3e5692bda6529f2cdc21e76c6bc0d34e209304ad796f83acb5905d4b84931d8eR93-R102

There's already a bit of a redirection, so it's probably good to take advantage of it.

Reply:
For some reason, the link doesn't show me where you're pointing. I'm assuming it's the `AST_Node.DEFMETHOD("evaluate", …)` definition, so I added a check there.